src/BigQueryClientWrapper.php

```diff
@@ -35,6 +35,30 @@ public function runQuery(JobConfigurationInterface $query, array $options = []):
         return parent::runQuery($query, $options);
     }
 
+    // Execute query and wait for it to finish
+    // Is recommended utilizing BigQueryClient::startQuery()
+    // as it provides better mechanisms for fine grained control over result polling.
+    public function executeQuery(QueryJobConfiguration $query): QueryResults
+    {
+        if ($this->runId !== '') {
+            /** @var QueryJobConfiguration $query */
+            $query = $query->labels(['run_id' => $this->runId]);
+        }
+        $job = $this->startQuery($query);
+
+        $retriesCount = 0;
+        do {
+            $waitSeconds = (int) min(pow(2, $retriesCount), 20);
+            sleep($waitSeconds);
```

Comment:
Asi v cajku radši bych teda viděl použitý přímo `UniformRandomBackOffPolicy` než něco takového, ale řekl bych že cajk.
Retry na reload na síťové problémy by měl řešit retry v clientu a stejně tak ten startQuery tak snad stím nebudů problémy.

Reply:
Možná bych to teda nacal do util ještě aby sa to mohlo přímo použit v `google-bigquery-transformation`

Reply:
Jo ty, to som chcel spravit vlastne, ale chcel som to vyskusat, tak ja to vycuknem este vedla.